pkg/operator/ceph/object/controller_test.go

```
@@ -754,6 +755,259 @@ func TestCephObjectStoreControllerMultisite(t *testing.T) {
+		}
+		r := getReconciler(objects)
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.False(t, calledCommitConfigChanges)
+		err = r.client.Get(ctx, req.NamespacedName, externalObjectStore)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete the same external store", func(t *testing.T) {
+		objects := []runtime.Object{
+			cephCluster,
+			externalObjectStore,
+			rgwAdminOpsUserSecret,
+		}
+		r := getReconciler(objects)
+
+		calledCommitConfigChanges = false
+
+		// no dependents
+		dependentsChecked := false
+		cephObjectStoreDependentsOrig := cephObjectStoreDependents
+		defer func() { cephObjectStoreDependents = cephObjectStoreDependentsOrig }()
+		cephObjectStoreDependents = func(clusterdCtx *clusterd.Context, clusterInfo *client.ClusterInfo, store *cephv1.CephObjectStore, objCtx *Context, opsCtx *AdminOpsContext) (*dependents.DependentList, error) {
+			dependentsChecked = true
+			return &dependents.DependentList{}, nil
+		}
+
+		err := r.client.Get(ctx, req.NamespacedName, externalObjectStore)
+		assert.NoError(t, err)
+		externalObjectStore.DeletionTimestamp = &metav1.Time{
+			Time: time.Now(),
+		}
+		err = r.client.Update(ctx, externalObjectStore)
+		assert.NoError(t, err)
+
+		// have to also track the same objects in the rook clientset
+		r.context.RookClientset = rookfake.NewSimpleClientset(externalObjectStore)
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, dependentsChecked)
+		assert.False(t, calledCommitConfigChanges)
+	})
+
+	t.Run("create an external object store with missing secret", func(t *testing.T) {
+		objects := []runtime.Object{
+			cephCluster,
+			externalObjectStore,
+		}
+		r := getReconciler(objects)
+		res, err := r.Reconcile(ctx, req)
+		assert.Error(t, err)
+		assert.False(t, res.Requeue)
+		assert.False(t, calledCommitConfigChanges)
+		err = r.client.Get(ctx, req.NamespacedName, externalObjectStore)
```

Sorry, I am a bit confused here, do `get` need to return the error here? Since  `externalObjectStore` reconcile failed. Same for 1003.

---

I think `get` returns `nil` as the error value here since `externalObjectStore` is still present in the runtime environment?

https://github.com/rook/rook/pull/9710/files#diff-10bcb1ddae3b1fddd221dda0765d8dc3987af28dae3a1944dbeaadc31da531fdR979

---

okay

---

What do we need this `get` for?

---

Redundant, removed.